### PR TITLE
add consul sync

### PR DESCRIPTION
**Pull request by impactCn**

### CODE REVIEW

File: lib/shenyu/register/consul.lua [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+
+function _M.init(conf)
+
+    _M.uri = conf.uri
+    _M.path = conf.path
+    _M.storage = conf.shenyu_storage
+
+    if 0 == ngx.worker.id() then
+        local ok, err = new_timer(5, _M:sync())
```
Comment (zhc00) [inline lib/shenyu/register/consul.lua]: sync should be a function pointer.

File: lib/shenyu/register/consul.lua [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+function _M:get_server_list()
+    local upstreams_str = _M.storage:get("server_list");
+    local tmp_upstreams = json.decode(upstreams_str);
+    return tmp_upstreams;
+end
+
+function _M:pick_and_set_peer()
+    local tmp_upstreams = _M.get_server_list();
+    local ip_port = tmp_upstreams[math.random(1, table.getn(tmp_upstreams))];
```
Comment (zhc00) [inline lib/shenyu/register/consul.lua]: `table.getn` is incorrect. refer to `register/balancer.lua`

Comment (impactCn) [reply lib/shenyu/register/consul.lua]: ok, let me see the writing on nacos.

**Comment by zhc00** [comment]
Please update the branch.

File: example/etcd/nginx.conf [hunk 3]
```diff
@@ -23,14 +23,14 @@ events {
...
 http {
-    lua_shared_dict shenyu_storage 1m;
+    lua_shared_dict shenyu_storage 10m;
 
     init_worker_by_lua_block {
         local register = require("shenyu.register.etcd")
         register.init({
             shenyu_storage = ngx.shared.shenyu_storage,
             balancer_type = "chash",
-            etcd_base_url = "http://127.0.0.01:2379",
```
Comment (zhc00) [inline example/etcd/nginx.conf:33]: typo, 127.0.0.1

File: lib/shenyu/register/consul.lua [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+_M._VERSION="0.1"
+
+local function sync()
+    local httpc = http.new()   
+
+    local resp, err = httpc:request_uri(_M.uri, {
+        method = "GET",  
+        path = _M.path,
+
```
Comment (zhc00) [inline lib/shenyu/register/consul.lua:40]: first, remove the empty line(L36)
then, maybe, let us keep the same argument list. I think the `path` could make users confusing.

File: lib/shenyu/register/consul.lua (hunk 2 above)
Comment (zhc00) [reply lib/shenyu/register/consul.lua]: let us use `lua-resty-balancer`

**Review by prFor — APPROVED** [review]

**Pull request merged**